### DISCUSSION TITLE
Should decomposition functions echo GHI in return?

### DISCUSSION THREAD

**Discussion by mikofski** [pvlib/pvlib-python#1676] [Ideas]
As [proposed](https://github.com/pvlib/pvlib-python/pull/1179#discussion_r583736392) in #1179 should decomposition functions like [Erbs](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.erbs.html) also echo GHI in the return?

**Comment by kandersolar** [comment]
I like the idea of the decomposition functions having a common return structure (kind of like #1553).  I'd want that structure to include (at least) all three of `ghi`, `dni`, and `dhi`.

One problem is that not all of those functions return frame-like structures which make it easy to add another column; `dirindex` and `dirint` just return dni by itself, so it would be breaking to change them to return multi-column structures.  I think it's probably worth it, but just wanted to point out that this change wouldn't be purely additive.

**Comment by wholmgren** [comment]
I think not. I would prefer to see `erbs` and `boland` return only `dni`. I'd also favor a little refactoring to create functions like `diffuse_fraction_erbs` and `diffuse_fraction_boland`.

**Reply by cwhanse** [reply]
Do you mean return only `dhi`? They are diffuse fraction models.

**Reply by wholmgren** [reply]
I had `dirint` on my mind so I was thinking `dni`. I would be fine with only `dhi`.

**Comment by cwhanse** [comment]
Maybe returning all of `ghi`, `dhi` and `dni` should be the job of a general function like `get_poa_components`

**Comment by adriesse** [comment]
> Maybe returning all of ghi, dhi and dni should be the job of a general function like get_poa_components

And then we could have some simpler base functions that just have the main model equations, like erbs mapping kt to df.
